core/db/src/main/java/org/opennms/netmgt/config/opennmsDataSources/JdbcDataSource.java

```diff
@@ -414,12 +420,29 @@ public void setUrl(final String url) {
     }
 
     /**
-     * Sets the value of field 'userName'.
+     * Sets the value of field 'rawUserName'.
      * 
-     * @param userName the value of field 'userName'.
+     * @param rawUserName the value of field 'rawUserName'.
      */
-    public void setUserName(final String userName) {
-        this.userName = userName;
+    public void setUserName(final String rawUserName) {
+        this.rawUserName = rawUserName;
     }
 
+    public String interpolateAttribute(final String value) {
+        return interpolateAttribute(value, "/opt/opennms/etc/scv.jce","QqSezYvBtk2gzrdpggMHvt5fJGWCdkRw");
+    }
+
+    public String interpolateAttribute(final String value, final String keystoreFile, final String password) {
+        final SecureCredentialsVault secureCredentialsVault = new JCEKSSecureCredentialsVault(keystoreFile, password);
+        final Interpolator.Result result = Interpolator.interpolate(value, new SecureCredentialsVaultScope(secureCredentialsVault));
+        return result.output;
+    }
+
+    public String getUserName() {
+        return interpolateAttribute(getRawUserName());
```

Comment:
What happens if there is a bad metadata expression ? 
We should catch exception from interpolation and throw relevant exception from here

Reply:
We are not throwing any exceptions. Any Metadata expression in the input string are replaced with the corresponding data in the output string. If this is not possible the the output string equals the input string. But we're not throwing any exceptions.